### PR TITLE
Add regressions for array sequence solver

### PR DESCRIPTION
**Pull request by ajreynol**
This closes the seqArray branch.

### CODE REVIEW

File: test/regress/regress0/seq/array/distinct-update.smt2 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+(assert (= y (seq.update x 0 (seq.unit a))))
+
+(assert (= z (seq.update x 0 (seq.unit b))))
+
+(assert (not (= a b)))
+(assert (= y z))
+(assert (> (str.len y) 0))
+
+(set-info :status unsat)
```
Comment (yoni206) [inline test/regress/regress0/seq/array/distinct-update.smt2:18]: Should this line be at the top?

File: test/regress/regress0/seq/rev.smt2 [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+(declare-fun A () (Seq Int))
+(declare-fun B () (Seq Int))
+(declare-const x Int)
+
+(assert (= B (seq.rev A)))
+(assert (and (<= 0 x) (< x (seq.len A))))
+(assert (not (= (seq.nth A x) (seq.nth B (- (- (seq.len A) 1) x)))))
+
+(set-info :status unsat)
```
Comment (yoni206) [inline test/regress/regress0/seq/rev.smt2:12]: Ditto

File: test/regress/regress0/seq/array/update-word-eq.smt2 [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+(declare-fun y () (Seq Int))
+(declare-fun z () (Seq Int))
+(declare-fun a () Int)
+(declare-fun b () Int)
+(declare-fun i () Int)
+
+(assert (= (seq.++ (seq.unit a) z) (seq.update x 0 (seq.unit b))))
+(assert (not (= a b)))
+
+(set-info :status unsat)
```
Comment (yoni206) [inline test/regress/regress0/seq/array/update-word-eq.smt2:14]: Ditto

**Review by yoni206 — APPROVED** [review]
Minor comments but LGTM

File: test/regress/regress0/seq/array/distinct-update.smt2 (hunk 1 above)
Comment (ajreynol) [reply test/regress/regress0/seq/array/distinct-update.smt2:18]: I don't think it matters, but I will move.

**Pull request merged**